zerver/lib/actions.py

```diff
@@ -2737,6 +2763,28 @@ def our_person(p):
                         bot.update(event['bot'])
 
         elif event['type'] == 'stream':
+            if event['op'] == 'create':
+                for stream in event['streams']:
+                    stream_name = stream['name']
+                    realm = get_realm('zulip.com')
+                    new_stream, created = create_stream_if_needed(realm, stream_name, stream['invite_only'])
```

Comment:
This code path should never create a stream -- if you read the flow, a stream event with 'op=create' is generated by the functions that create a stream _after_ they've already created the stream.  The events are to notify browsers about the new stream.

If you need to fetch the stream object, you can use `get_stream`.  But `apply_events` should never contain or call anything that results in a database write (such as create_stream_if_needed).